src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/DirectoryBasedStoreProvider.cs

```diff
@@ -298,24 +302,7 @@ public void Add(ICertificatePal certPal)
                     stream.Write(pkcs12, 0, pkcs12.Length);
                 }
 
-#if DEBUG
-                // Verify that we're creating files with u+rw and o-rw, g-rw.
-                const Interop.Sys.Permissions requiredPermissions =
-                    Interop.Sys.Permissions.S_IRUSR |
-                    Interop.Sys.Permissions.S_IWUSR;
-
-                const Interop.Sys.Permissions forbiddenPermissions =
-                    Interop.Sys.Permissions.S_IROTH |
-                    Interop.Sys.Permissions.S_IWOTH |
-                    Interop.Sys.Permissions.S_IRGRP |
-                    Interop.Sys.Permissions.S_IWGRP;
-
-                Interop.Sys.FileStatus stat;
-
-                Debug.Assert(Interop.Sys.Stat(destinationFilename, out stat) == 0);
-                Debug.Assert((stat.Mode & (int)requiredPermissions) == (int)requiredPermissions, "Created PFX has insufficient permissions to function");
-                Debug.Assert((stat.Mode & (int)forbiddenPermissions) == 0, "Created PFX has too broad of permissions");
-#endif
+                EnsureFilePermissions(destinationFilename, userId);
```

Comment:
For the case where the file is being created anew, should we ensure the permissions are correct before we write the cert data to the file? e.g.
```C#
if (existingFilename != null)
{
    destinationFilename = existingFilename;
}
else
{
    destinationFilename = findOpenSlot ?
        FindOpenSlot(thumbprint) :
        Path.Combine(_storePath, thumbprint + PfxExtension);

    File.Open(destinationFilename, FileMode.CreateNew).Dispose();
}

EnsureFilePermissions(destinationFilename, userId);

using (FileStream stream = File.Open(destinationFilename, FileMode.Open))
{
    byte[] pkcs12 = copy.Export(X509ContentType.Pkcs12);
    stream.Write(pkcs12, 0, pkcs12.Length);
}
```
?